lectures/tools_and_techniques/ar1_processes.md

```diff
@@ -0,0 +1,575 @@
+
+```{math}
+:label: ar1_ergo
+
+\frac{1}{m} \sum_{t = 1}^m h(X_t)  \to
+\int h(x) \psi^*(x) dx
+    \quad \text{as } m \to \infty
+```
+
+whenever the integral on the right hand side is finite and well defined.
+
+Notes:
+
+* In {eq}`ar1_ergo`, convergence holds with probability one.
+* The textbook by {cite}`MeynTweedie2009` is a classic reference on ergodicity.
+
+For example, if we consider the identity function $h(x) = x$, we get
+
+$$
+\frac{1}{m} \sum_{t = 1}^m X_t  \to
+\int x \psi^*(x) dx
+    \quad \text{as } m \to \infty
+$$
+
+In other words, the time series sample mean converges to the mean of the
+stationary distribution.
+
+As will become clear over the next few lectures, ergodicity is a very
+important concept for statistics and simulation.
+
+## Exercises
+
+% (ar1p_ex1)=
+% ### Exercise 1
+
+% Let $k$ be a natural number.
+
+% The $k$-th central moment of a  random variable is defined as
+
+% $$
+% M_k := \mathbb E [ (X - \mathbb E X )^k ]
+% $$
+
+% When that random variable is $N(\mu, \sigma^2)$, it is known that
+
+% $$
+% M_k =
+% \begin{cases}
+%     0 & \text{ if } k \text{ is odd} \\
+%     \sigma^k (k-1)!! & \text{ if } k \text{ is even}
+% \end{cases}
+% $$
+
+% Here $n!!$ is the double factorial.
+
+% According to {eq}`ar1_ergo`, we should have, for any $k \in \mathbb N$,
+
+% $$
+% \frac{1}{m} \sum_{t = 1}^m
+%     (X_t - \mu^* )^k
+%     \approx M_k
+% $$
+
+% when $m$ is large.
+
+% Confirm this by simulation at a range of $k$ using the default parameters from the lecture.
+
+(ar1p_ex2)=
+### Exercise 1
+
+Write your own version of a one dimensional [kernel density
+estimator](https://en.wikipedia.org/wiki/Kernel_density_estimation),
+which estimates a density from a sample.
+
+Write it as a class that takes the data $X$ and bandwidth
+$h$ when initialized and provides a method $f$ such that
+
+$$
+f(x) = \frac{1}{hn} \sum_{i=1}^n
+K \left( \frac{x-X_i}{h} \right)
+$$
+
+For $K$ use the Gaussian kernel ($K$ is the standard normal
+density).
+
+Write the class so that the bandwidth defaults to Silverman’s rule (see
+the “rule of thumb” discussion on [this
+page](https://en.wikipedia.org/wiki/Kernel_density_estimation)). Test
+the class you have written by going through the steps
+
+1. simulate data $X_1, \ldots, X_n$ from distribution $\phi$
+1. plot the kernel density estimate over a suitable range
+1. plot the density of $\phi$ on the same figure
+
+for distributions $\phi$ of the following types
+
+- [beta
+  distribution](https://en.wikipedia.org/wiki/Beta_distribution)
+  with $\alpha = \beta = 2$
+- [beta
+  distribution](https://en.wikipedia.org/wiki/Beta_distribution)
+  with $\alpha = 2$ and $\beta = 5$
+- [beta
+  distribution](https://en.wikipedia.org/wiki/Beta_distribution)
+  with $\alpha = \beta = 0.5$
+
+Use $n=500$.
+
+Make a comment on your results. (Do you think this is a good estimator
+of these distributions?)
+
+(ar1p_ex3)=
+### Exercise 2
```

Comment:
Is this one able to be done without special packages?  If so, lets put it back in (and rename to excercise 1)?